### PR TITLE
Use datadeps for AvgPerceptronTagger instead storing weights inside package

### PR DESCRIPTION
**Pull request by Ayushk4**
I have switched to using datadeps instead of storing weights locally, similar to the NER API.
As of now, I am directly taking it from version control history, but perhaps it could be released as an asset.

### CODE REVIEW

File: src/avgPerceptronTagger_DataDeps.jl [hunk 1]
```diff
@@ -0,0 +1,14 @@
+using DataDeps
```
Comment (aviks) [inline src/avgPerceptronTagger_DataDeps.jl]: add this to the averagePerceptronTagger.jl file? Seems very little code for a new file.

File: src/TextAnalysis.jl [hunk 2]
```diff
@@ -60,6 +61,10 @@ module TextAnalysis
 
     export CRF, viterbi_decode, crf_loss
 
+    function __init__()
+        include(joinpath(@__DIR__, "avgPerceptronTagger_DataDeps.jl"))
```
Comment (aviks) [inline src/TextAnalysis.jl]: don't do an `include` in the `__init__`. make it a function, and call the function.

**Comment by Ayushk4** [comment]
Thanks for the review, I will make the changes ASAP.

File: src/avgPerceptronTagger_DataDeps.jl (hunk 1 above)
Comment (Ayushk4) [reply src/avgPerceptronTagger_DataDeps.jl]: Done.

File: src/TextAnalysis.jl (hunk 2 above)
Comment (Ayushk4) [reply src/TextAnalysis.jl]: Done.

**Comment by Ayushk4** [comment]
@aviks I have made the suggested changes. 

I also provided `POS Tagger` support over various `Document` types and `String` types, updated tests, docstrings and Documentation for the same.

Please review this (since there will be some merge conflicts with [#167_Comment](https://github.com/JuliaText/TextAnalysis.jl/pull/167#discussion_r312684612) )

**Comment by Ayushk4** [comment]
As of now, I am taking the weights from the version history [(link)](https://github.com/JuliaText/TextAnalysis.jl/blob/cd30c10e9a23ca4369b6753d582eda10a9a86021/src/averagePerceptronTagger.jl#L13). It may be neater to release the file as an asset on GitHub [(similar to MetalHead.jl)](https://github.com/FluxML/Metalhead.jl/releases/tag/v0.1.1). 

Also, various other NLP libraries like SpaCy use avg perceptron tagger from POS. Maybe the model weights from these libraries along with one we currently provide could be tested, and then the best one could be made available from this package. Once this is done, we can release the weights.

**Comment by aviks** [comment]
> release the file as an asset on GitHub

I have added the file (zipped) into a release on this package.

https://github.com/JuliaText/TextAnalysis.jl/releases/download/v0.6.0/pretrainedMod.bson.zip

**Comment by Ayushk4** [comment]
I have changed the link to the weights.

**Pull request merged**